lms/djangoapps/course_api/api.py

```diff
@@ -0,0 +1,58 @@
+"""
+Course API
+"""
+
+from django.contrib.auth.models import User
+from lms.djangoapps.courseware.courses import (
+    get_courses,
+    get_course_by_id
+)
+from opaque_keys.edx.keys import CourseKey
+
+from .serializers import CourseSerializer
+
+
+def _has_permission(requesting_user, username):
+    """
+    Returns true if `requesting_user` has permission to access the user
+    identified by `username`.
+
+    Arguments:
+        requesting_user (User): The user requesting permission to view another
+        username (string): The name of the user `requesting_user` would like
+        to access.
+
+    Return value:
+        Boolean
+    """
+    if not username:
+        return False
+    return requesting_user.is_staff or requesting_user.username == username
+
+
+def course_detail(course_key_string, request):
```

Comment:
The reason I put it at the end is that unlike for a view (where the request is a central part of the how the view does its work), the api only takes the request incidentally because of how we pull the course description.  If we refactor the course description to be part of the `CourseOverview`, the `request` will go away altogether.  

The primary contract of the API is "give me a course_key, and I'll give you a representation of the course.  Oh and by the way, due to implementation details I need a request object to get the job done."  That's different from the view, which has the contract of "Make an HTTP request to me, and I will give you an HTTP response."  That's why the signature is different.